File: interface/src/main/java/cn/itcast/core/service/BrandService.java
package cn.itcast.core.service;

import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.Map;
@Repository
public interface BrandService {
    //查询所有
    List<Brand> queryAll();
    //分页查询
    PageResult findPage(Integer page,Integer rows);
    //新增品牌
    void add(Brand brand);
    //查询一条
    Brand findOne(Long id);
    //更新品牌
    void update(Brand brand);
    //批量删除
    void delete(Long[] ids);
    //多条件分页查询
    PageResult search(Integer page,Integer rows,Brand brand);
    //获取品牌列表数据格式是封装的map对象
    List<Map> selectOptionList();

    //修改状态
    void updateStatus(Long[] ids, String status);
}

File: web-manager/src/main/webapp/js/service/brandService.js
app.service('brandService',function ($http) {
    //查询所有并展示
    this.findAll=function () {
       return $http.get('../brand/findAll.do');
    }
    //分页查询
    this.findPage=function (page, rows) {
        return $http.get('../brand/findPage.do?page='+page+'&rows='+rows);
    }
    //添加
    this.add=function (entity) {
        return $http.post('../brand/add.do',entity);
    }
    //更新
    this.update=function (entity) {
        return $http.post('../brand/update.do',entity);
    }
    //根据主键查询一条(修改时回显)
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id='+id);
    }
    //批量删除
    this.dele=function (ids) {
        return $http.get('../brand/delete.do?ids='+ids);
    }
    //条件查询带分页
    this.search=function (page, rows, searchEntity) {
        return $http.post('../brand/search.do?page='+page+'&rows='+rows,searchEntity);
    }
    //获取品牌列表数据格式是封装的map对象
    this.selectOptionList = function () {
        return $http.get('../brand/selectOptionList.do');
    }
    this.updateStatus = function(ids,status){
        return $http.get('../brand/updateStatus.do?ids='+ids+"&status="+status);
    }
})